api/src/main/java/io/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -44,13 +47,26 @@ public FileIteratingFirehose(
   {
     this.lineIterators = lineIterators;
     this.parser = parser;
+    final ParseSpec parseSpec = parser.getParseSpec();
+    if (parseSpec instanceof CSVParseSpec) {
```

Comment:
I think the rationale was that we want to avoid instanceof (since it's brittle) and we want to avoid changing the signature of the `Parser.parse` method (since it's in druid-api and those interfaces are meant to be stable). So returning nulls to signal "skip me" accomplishes those.

I think it's ok for a Parser to behave that way, since it's supposed to throw ParseException if there was really a parse error. Returning null to signal "skip me" or seems reasonable.

I'm definitely open to other suggestions though. If you think it's best to change `Parser.parse` we could perhaps deprecate it and add a new method that returns something like `ParseResult`.